File: data structures and algorithms/java/05_036.java
import java.util.*;
import java.util.stream.*;
import java.io.*;
import java.math.*;

public class Main {
    public static void main(String[] args) {
        // Main method code goes here
    }
    
    public class Graph {
        private List<Node> nodes;
        
        public void removeNode(Node node) {
            // First attempt - not very efficient!
            for (Node currentNode : this.nodes) {
                for (Edge edge : currentNode.edges) {
                    if (edge.connectsTo(node)) {
                        currentNode.edges.remove(edge);
                    }
                }
            }
            this.nodes.remove(node);
        }
    }
}

File: data structures and algorithms/cs/05_036.cs
using System;
using System.Collections.Generic;

public class MainClass
{
   public static void Main(string[] args)
   {
      // Main method code goes here
   }

   public class Graph
   {
      private List<Node> nodes;

      public void RemoveNode(Node node)
      {
         // First attempt - not very efficient!
         foreach (var currentNode in nodes)
         {
            currentNode.Edges.RemoveAll(edge => edge.ConnectsTo(node));
         }
         nodes.Remove(node);
      }

      public void ViewGraph()
      {
         foreach (Node node in nodes)
         {
            Console.Write(node + ": ");
            foreach (Edge edge in node.Edges)
            {
               Console.Write(edge + " ");
            }
            Console.WriteLine();
         }
      }
   }

   public class Node
   {
      public List<Edge> Edges { get; set; }
   }

   public class Edge
   {
      public bool ConnectsTo(Node node)
      {
         // Implementation goes here
         return false;
      }
   }
}
